atmo/jobs/models.py

```diff
@@ -204,14 +187,18 @@ def run(self):
             is_public=self.is_public,
             job_timeout=self.job_timeout,
         )
-        self.last_run_date = timezone.now()
-        self.update_status()
-        self.save()
+        # Create new job history record.
+        run = self.runs.create(
+            spark_job=self,
+            jobflow_id=jobflow_id,
+            scheduled_date = timezone.now(),
+        )
+        run.update_status()
```

Comment:
Once the job run has been created we need to invalidate the cached_property of `self.latest_run` since it could otherwise lead to race conditions